app/views/appointments_adm.py

```diff
@@ -0,0 +1,90 @@
+import json
+from datetime import datetime
+
+from flask import render_template, request, Blueprint, flash, redirect, url_for
+from flask_login import login_required
+
+from app.models import appointments as appt_model
+from app.models import services as service_model
+from app.models import shops as shop_model
+
+
+bp = Blueprint("appointments", __name__)
+
+
+@bp.route("/admin/appointments")
+@login_required
+def admin_appointments_management():
+    data = appt_model.all()
+    events = []
+    for appt in data:
+        event = {
+            "id": appt["id"],
+            "title": appt["name"],
+            "start": str(appt["timeslot"]),
+            "allDay": True,
+        }
+        if appt["timeslot"] < datetime.now():
+            event["backgroundColor"] = "#989898"
+            event["borderColor"] = "#989898"
+        events.append(event)
+    return render_template(
+        "admin/appointments/index.html",
+        active_nav="appointments",
+        appointments=json.dumps(events),
+    )
+
+
+@bp.route("/admin/appointments/view/upsert", methods=["POST"])
+def admin_render_upsert_form():
+    shops = shop_model.all()
```

Comment:
curious, why we are pulling all the shops in here? is it possible to get the shop id from the form?

Reply:
That route is using to render the form.
I get all shops and services records to populate shop and service dropdown options in upsert_form template